src/config/load.c

```diff
@@ -66,18 +65,12 @@ cfgLoadUpdateOption(void)
     FUNCTION_LOG_VOID(logLevelTrace);
 
     // Set default for repo-host-cmd
-    if (cfgOptionTest(cfgOptRepoHost) && cfgOptionSource(cfgOptRepoHostCmd) == cfgSourceDefault)
+    if (cfgOptionValid(cfgOptRepoHostCmd))
         cfgOptionDefaultSet(cfgOptRepoHostCmd, VARSTR(cfgExe()));
 
     // Set default for pg-host-cmd
     if (cfgOptionValid(cfgOptPgHostCmd))
-    {
-        for (unsigned int optionIdx = 0; optionIdx < cfgOptionGroupIdxTotal(cfgOptGrpPg); optionIdx++)
-        {
-            if (cfgOptionTest(cfgOptPgHost + optionIdx) && cfgOptionSource(cfgOptPgHostCmd + optionIdx) == cfgSourceDefault)
-                cfgOptionDefaultSet(cfgOptPgHostCmd + optionIdx, VARSTR(cfgExe()));
-        }
-    }
+        cfgOptionDefaultSet(cfgOptPgHostCmd, VARSTR(cfgExe()));
```

Comment:
1) In cfgOptionDefaultSet(), we're setting the default without even checking the source - so if a user set the option then we'd just overwrite it. 

2) And I'm guessing we're no longer checking to see if the pg-host option is valid and has a value because, what, cfgOptPgHostCmd is independent of that so we no longer care?

Reply:
Github won't let me reply to the comment above, so I'll replay to both comments here.

`cfgOptionDefault()` and `cfgOptionDefaultValue()` are only used for help. There's probably no reason for them to store the value at all but this is how it was done before. This does not *set* a default, it just stores a copy of the already existing default, pulled from `cfgDefOptionDefault()`. If I had my way both of these functions would be gone and moved to help but it's not practical in this commit

We can clarify that these functions are used only for help but I'd rather not spend too much time on it right now.

`cfgOptionDefaultSet()` is used to set a *new* default and is used in just a few places.

` we're setting the default without even checking the source - so if a user set the option then we'd just overwrite it.`

Line 637 checks the source so it is only overwritten if it is the default.

Reply:
Then I would like to clarify that restore is just checking the target action for help: https://github.com/pgbackrest/pgbackrest/blob/dev-mr-010-option-idx/src/command/restore/restore.c#L1412 

and that the setting of the configLocal default value in https://github.com/pgbackrest/pgbackrest/blob/dev-mr-010-option-idx/src/config/config.c#L608 is not really setting anything although it looks like it is (or could) when restore calls it

and that line 632 https://github.com/pgbackrest/pgbackrest/blob/dev-mr-010-option-idx/src/config/config.c#L632 just overwrites without checking the source - this function can be called for options that are not indexed and there isn't really any reason not to put in a check that point.

Reply:
This is all true, but the functionality has remained the same (as far as I know) so I'd prefer to leave this as is for now.

In general I'd like for help to deal with defaults on it's own, while also respecting any that were explicitly set after parsing. I think restore could just use `cfgOptionSource() == cfgSourceDefault` but that requires some thought and I don't want to make it part of this commit.

Reply:
understood. I added a small comment in a separate commit so it is easier to revert if needed: https://github.com/pgbackrest/pgbackrest/pull/1221/commits/d808efcbea8dd8023da96a2b72bf616c6ca324da

Reply:
Comment, looks fine. I removed cfgOptionDefault() from restore in 45138219595 to make it true.